### PR TITLE
Fix bug in student's T quantile for denormalized degrees_of_freedom.

### PR DESCRIPTION
**Pull request by jzmaddock**
See https://github.com/boostorg/math/issues/406

### CODE REVIEW

File: include/boost/math/special_functions/detail/ibeta_inverse.hpp [hunk 1]
```diff
@@ -673,7 +673,28 @@ T ibeta_inv_imp(T a, T b, T p, T q, const Policy& pol, T* py)
...
+      try {
+         bet = boost::math::beta(a, b, pol);
+      }
+      catch (const std::runtime_error&)
+      {
+         overflow = true;
+      }
+      if (overflow || !(boost::math::isfinite)(bet))
+      {
+         xg = exp((boost::math::lgamma(a + 1, pol) + boost::math::lgamma(b + 1) - boost::math::lgamma(a + b + 1) + log(p) - log(b) + log(a + b)) / a);
```
Comment (evanmiller) [inline include/boost/math/special_functions/detail/ibeta_inverse.hpp:694]: Shouldn't the 2nd and 3rd calls to `lgamma` here include the policy as the 2nd argument?

**Comment by jzmaddock** [comment]
>Shouldn't the 2nd and 3rd calls to lgamma here include the policy as the 2nd argument?

Doh!  Yes.

File: include/boost/math/special_functions/detail/ibeta_inverse.hpp [hunk 2]
```diff
@@ -673,7 +673,34 @@ T ibeta_inv_imp(T a, T b, T p, T q, const Policy& pol, T* py)
...
+      try {
+         bet = boost::math::beta(a, b, pol);
+      }
+      catch (const std::runtime_error&)
+      {
+         overflow = true;
+      }
+      if (overflow || !(boost::math::isfinite)(bet))
+      {
+         xg = exp((boost::math::lgamma(a + 1, pol) + boost::math::lgamma(b + 1, pol) - boost::math::lgamma(a + b + 1, pol) + log(p) - log(b) + log(a + b)) / a);
```
Comment (evanmiller) [inline include/boost/math/special_functions/detail/ibeta_inverse.hpp:700]: Maybe I'm missing something, but why not reduce

`lgamma(b+1) - log(b) - lgamma(a+b+1) + log(a+b)`

To just `lgamma(b) - lgamma(a+b)`

?

Comment (jzmaddock) [reply include/boost/math/special_functions/detail/ibeta_inverse.hpp:700]: The branch deals with a,b < 1 and the problem occurs when one of a,b is so small that 1/a (for example) overflows, so I had issues with lgamma overflowing as well.... though in point of fact I guess that may be a bug as well :(

Comment (jzmaddock) [reply include/boost/math/special_functions/detail/ibeta_inverse.hpp]: It was a bug in lgamma: there was actually a fix for this already in place, it just didn't quite go far enough.

**Pull request merged**